internal/pkg/correlation/middleware.go

```diff
@@ -17,9 +17,13 @@ func ManageHeader(next http.Handler) http.Handler {
 		if correlationID == "" {
 			correlationID = uuid.New().String()
 		}
+		// lint:ignore SA1029 legacy
+		// nolint:staticcheck // SA1004 legacy
```

Comment:
Intersting that it doen't think you should use a key of type string when the example in GoDoc use string. :-)
https://pkg.go.dev/context#example-WithValue
But they do create a new type to obscure that it is a string. Humm, bet this keeps the linter happy.

SA1004 doesn't make sense here: https://staticcheck.io/docs/checks#SA1004
Does `nolint:staticcheck` disable the `staticcheck` for this line? if so, do we really need `lint:ignore SA1029 legacy` also

Reply:
It just disables staticcheck for this line

I fixed this to read //nolint:staticcheck // See golangci-lint #741

Its a known issue https://github.com/golangci/golangci-lint/issues/741

Reply:
> Intersting that it doen't think you should use a key of type string when the example in GoDoc use string. :-) https://pkg.go.dev/context#example-WithValue But they do create a new type to obscure that it is a string. Humm, bet this keeps the linter happy.

It's more than that.  If you have a web server with a very deep HTTP pipeline, you can get collisions when people stuff things like "id" = "1234" into the context because "id" for one module is not the same as "id" for another module.  By wrapping it in a typed struct, you mostly eliminate the possibility for the collision because the type is going to be scoped by the package so even if the field is the same, across packages the full type name of the field will be different.

Reply:
I think it's ok if we just grandfather this one in.  We only have two.